kit/inputs/src/UIAmountInput/hooks.tsx

```diff
@@ -148,9 +148,13 @@ export function useUIAmountInputChildren(
         return materialTextViewChildren;
     }
 
-    if (!hideClearButton && inputHasValue && (isFocused || isHovered)) {
+    if (hideClearButton) {
+        return undefined;
+    }
+
+    if (inputHasValue && (isFocused || isHovered)) {
         return <MaterialTextViewClearButton clear={clear} />;
     }
 
-    return undefined;
+    return <MaterialTextViewClearButton hiddenButton />;
```

Comment:
I suggest to add comments for each conditions in order to recall once we need it, why the logic for this component is this one. Since it differs to the logic of UIMaterialTextView component